src/utils/baseClass.js

```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import shallowCompare from 'react-addons-shallow-compare';
+
+class BaseClass extends (React.PureComponent || React.Component) {
+  [React.PureComponent || 'shouldComponentUpdate'](nextProps, nextState) {
```

Comment:
I'm not sure I fully understand what this will do in the case that `React.PureComponent` is defined

Reply:
shouldn't it actually be `!React.PureComponent &&` so that it injects the `shouldComponentUpdate` when PureComponent isn't available.

Reply:
aha, yes, that's better. coding is hard.